examples/amp-story/lcp.html

```diff
@@ -0,0 +1,107 @@
+  <style amp-custom>
+    * {
+      box-sizing: border-box;
+    }
+    amp-story-page {
+      font-family: 'Helvetica Neue', sans-serif;
+      color: white;
+      background: black;
+    }
+    amp-story p {
+      line-height: 1.5;
+    }
+    .bold {
+      font-weight: bold;
+    }
+    .bottom {
+      align-content: end;
+    }
+    .last {
+      padding-bottom: 8.83vh;
+    }
+    .byline {
+      letter-spacing: 1.28px;
+    }
+    img {
+      bottom: 35% !important;
+    }
+  </style>
+</head>
+  <body>
+    <amp-story standalone id="cats"
+        title="LCP test" publisher="The AMP team"
+        publisher-logo-src="./img/amplogo.png"
+        poster-portrait-src="./img/overview.jpg">
+
+      <amp-story-page id="page-1" title="Lorem ipsum dolor sit amet">
+        <amp-story-grid-layer template="fill">
+          <amp-img
+            id="img1"
+            width="400"
+            height="750"
+            src="https://images.unsplash.com/photo-1515705576963-95cad62945b6?w=1750&q=100&fm=webp"
+            layout="fill">
+          </amp-img>
+        </amp-story-grid-layer>
+        <amp-story-grid-layer template="vertical" class="bottom">
+          <p class="bold">LCP for images</p>
+          <p class="byline">load=first</p>
+          <p class="byline">compressed=true</p>
+        </amp-story-grid-layer>
+      </amp-story-page>
+
+      <amp-story-page id="page-2">
+        <amp-story-grid-layer template="fill">
+          <amp-img
+            id="img2"
+            width="400"
+            height="750"
+            src="https://images.unsplash.com/photo-1502318217862-aa4e294ba657?w=1080&q=100&fm=webp"
+            layout="fill">
+          </amp-img>
+        </amp-story-grid-layer>
+      </amp-story-page>
+
+      <amp-story-page id="page-3">
+        <amp-story-grid-layer template="fill">
+          <amp-img
+            id="img3"
+            width="400"
+            height="750"
+            src="https://images.unsplash.com/photo-1513809491260-0e192158ae44?w=1080&q=100&fm=webp"
+            layout="fill">
+          </amp-img>
+        </amp-story-grid-layer>
+      </amp-story-page>
+    </amp-story>
+    <script>
+      const loadAll = location.hash.includes('load=all');
+      (self.AMP = self.AMP || []).push(function(AMP) {
+        AMP.toggleExperiment('amp-story-load-first-page-only', !loadAll);
+      });
+      const bylines = document.querySelectorAll('.byline');
+      if (loadAll) {
+        bylines[0].textContent = 'load=all';
+      }
+      const compress = location.hash.includes('compressed=true');
+      const quality = compress ? 30 : 100;
+      document.querySelectorAll('amp-img').forEach((img) => {
+        img.setAttribute('src', img.getAttribute('src').replace('100', quality));
+      })
+      bylines[1].textContent = 'compressed=' + compress;
```

Comment:
I honestly think using a non valid AMP document that replaces some content after the DOM is parsed will cause side effects in LCP measurements.

Reply:
I have not seen this be a concern for calculating the LCP as AMP parses the elements after this script runs, so the LCP is not affected by this bit of JS running (similar to how AMP experiments are parsed before AMP runs, it will even run before the scripts get fetched). However, if you're still concerned, I can create many example stories with combinations of this, just that this approach is easier to test / modify and provides necessary test scenarios.

Reply:
I just think we should never introduce differences from the prod environment when trying to estimate savings on the prod environment. A few weeks ago we saw that your test file was impacting the metrics, even if it doesn't happen today this could happen at any time with any update.